### DISCUSSION TITLE
Setting default Jira URL and point values for an entire poinz instance

### DISCUSSION THREAD

**Discussion by cdwillie76** [Zuehlke/poinz#332] [General]
Is there a way to make a local code change so a poinz instance has the following defaults set/changed
- Set the Jira URL to be a default value for all users
- Change the default values for the card points

I tried making some code changes where I "thought" those changes could be made but it didn't work.

Thanks!
Chris

**Comment by xeronimus** [comment]
Hi @cdwillie76 ,

Thanks for your input!
The default values for these two configrations are currently hardcoded. This means, if you change them, you need to re-build the backend. (or at least restart the backend, if you are on your dev-environment).


### URL
The"jira url"  is called "issueTrackingUrl" and is a property on the room. It is set, each time a new room is created. 
You can edit it in the following file: [/server/src/eventHandlers/roomCreated.js](https://github.com/Zuehlke/poinz/blob/master/server/src/eventHandlers/roomCreated.js)

```javascript
function roomCreatedEventHandler(room) {
  return {
    id: room.id,
    users: [],
    stories: [],
    created: Date.now(),
    autoReveal: true,
    withConfidence: false,
    issueTrackingUrl: "https://www.my-jira-instance.com/path/to-issues/{ISSUE}"   // <-- set your url here.  the current default value is undefined
  };
}
```

### Cards
The default cards can be edited here: [/server/src/defaultCardConfig.js](https://github.com/Zuehlke/poinz/blob/master/server/src/defaultCardConfig.js)

```javascript
// change existing lines (value/label/color), remove lines or add new ones.
// just make sure that the "values" are unique!
const defaultCardConfig = [
  {label: '?', value: -2, color: '#bdbfbf'},
  {label: '1/2', value: 0.5, color: '#667a66'},
  {label: '1', value: 1, color: '#839e7a'},
  {label: '2', value: 2, color: '#8cb876'},
  {label: '3', value: 3, color: '#96ba5b'},
  {label: '5', value: 5, color: '#b6c76b'},
  {label: '8', value: 8, color: '#c9c857'},
  {label: '13', value: 13, color: '#d9be3b'},
  {label: '21', value: 21, color: '#d6cda1'},
  {label: '34', value: 34, color: '#9fa6bd'},
  {label: '55', value: 55, color: '#6a80ab'},
  {label: 'BIG', value: -1, color: '#1d508f'}
];
```

**Reply by cdwillie76** [reply]
That worked!  Thank you very much.  I was able to change 'BIG' to be the infinity symbol by putting in the value of '\u221E'.
